### PR TITLE
Add level1c runner

### PR DESCRIPTION
**Pull request by adybbroe**
This PR adds another runner to generate the level-1c file input which is needed by the NWCSF/PPS. From version 2021 and onwards the PPS package will require a standard (PPS) netCDF/CF level-1c format as input. This level-1c formatting is done with Satpy via the level1c4pps package (not Pytroll).

 - [ ] Closes #xxxx 
 - [x] Tests added 
 - [x] Tests passed: Passes ``pytest nwcsafpps_runner`` 
 - [x] Passes ``flake8 nwcsafpps_runner`` 
 - [ ] Fully documented

### CODE REVIEW

File: nwcsafpps_runner/l1c_processing.py [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+    if not epi_files:
+        LOG.warning("EPI file is missing...")
+        return []
+
+    return level1_files
+
+
+def check_message_okay(msg):
+    """Check that the message is okay and has the necessary fields."""
+
```
Comment (mraspaud) [inline nwcsafpps_runner/l1c_processing.py:174]:
```suggestion
```

File: nwcsafpps_runner/l1c_processing.py [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+        LOG.warning("PRO file is missing...")
+        return []
+    if not epi_files:
+        LOG.warning("EPI file is missing...")
+        return []
+
+    return level1_files
+
+
+def check_message_okay(msg):
```
Comment (mraspaud) [inline nwcsafpps_runner/l1c_processing.py:172]: what does it mean that the message is okay? could this function name be more precise?

File: nwcsafpps_runner/l1c_processing.py [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+        return False
+
+    if ('platform_name' not in msg.data or 'start_time' not in msg.data):
+        LOG.error("Message is lacking crucial fields...")
+        return False
+
+    return True
+
+
+def check_service_name_okay(service_name):
```
Comment (mraspaud) [inline nwcsafpps_runner/l1c_processing.py:189]: maybe `check_service_is_supported`?

**Review by mraspaud — COMMENTED** [review]
That's a big PR, thanks for putting in the effort!

I think generally that that the code could be made easier to follow with a few changes, see inline comments.

File: bin/level1c_runner.py [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+from posttroll.publisher import Publish
+
+from nwcsafpps_runner.logger import setup_logging
+from nwcsafpps_runner.message_utils import publish_l1c, prepare_l1c_message
+from nwcsafpps_runner.l1c_processing import L1cProcessor
+
+LOG = logging.getLogger('l1c-runner')
+
+
+def l1c_runner(config_filename, service_name):
```
[the remark this answers is not in the document]
Comment (adybbroe) [reply bin/level1c_runner.py:39]: Good question, it is not! 
I inherited this setup from somewhere else. Should have done it differently I guess

File: nwcsafpps_runner/config.py [hunk 5]
```diff
@@ -26,13 +26,43 @@
 import os
 import socket
 
+import yaml
+try:
+    from yaml import UnsafeLoader
+except ImportError:
+    from yaml import Loader as UnsafeLoader
```
[the remark this answers is not in the document]
Comment (adybbroe) [reply nwcsafpps_runner/config.py:33]: No we don't. Good point, thanks.

File: nwcsafpps_runner/l1c_processing.py (hunk 2 above)
Comment (adybbroe) [reply nwcsafpps_runner/l1c_processing.py:171]: The thing it checks mor ethan one thing, which indicates I should do more functions. I will think about it...

File: nwcsafpps_runner/l1c_processing.py (hunk 3 above)
Comment (adybbroe) [reply nwcsafpps_runner/l1c_processing.py:187]: Yes, better! Thanks!

File: bin/level1c_runner.py [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+
+    l1c_proc = L1cProcessor(config_filename, service_name)
+    publish_name = service_name + '-runner'
+
+    with Subscribe('', l1c_proc.subscribe_topics, True) as sub:
+        with Publish(publish_name, 0) as publisher:
+            while True:
+                for msg in sub.recv():
+                    l1c_proc.initialise(service_name)
+                    status = l1c_proc.run(msg)
```
[the remark this answers is not in the document]
Comment (adybbroe) [reply bin/level1c_runner.py:51]: Yes, right! I will fix.

File: nwcsafpps_runner/config.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (adybbroe) [reply nwcsafpps_runner/config.py]: Fixed.

File: nwcsafpps_runner/l1c_processing.py (hunk 3 above)
Comment (adybbroe) [reply nwcsafpps_runner/l1c_processing.py]: Fixed.

**Comment by adybbroe** [comment]
@mraspaud PR is improved I think. Good enough?

**Comment by mraspaud** [comment]
I would just like to have the stopping mechanism before merging this if possible.

**Comment by adybbroe** [comment]
> I would just like to have the stopping mechanism before merging this if possible.

@mraspaud I made an attempt, without re-factoring too much. Not so very comfortable with these kind of mechanisms. Was it something like this you had in mind?

**Comment by mraspaud** [comment]
Sorry I didn't have time to review this PR this afternoon.
Just a note regarding the loop variable. The way you implemented it, you don't need to make it global, it's enough that it is defined in the context of the outer function.

**Pull request merged**